src/docbuilder/rustwide_builder.rs

```diff
@@ -289,125 +304,133 @@ impl RustwideBuilder {
         }
 
         let mut build_dir = self.workspace.build_dir(&format!("{}-{}", name, version));
-        build_dir.purge()?;
+        build_dir.purge().map_err(FailureError::compat)?;
```

Comment:
There's an awful lot of `FailureError::compat` in this file - could we maybe return a `failure::Error` from `build_package()` so we only have to convert it once? Or does that break backtraces because they're not stable in std?

Reply:
there I'm really not sure what's best, could be come tricky. 

This probably would mean changing the return types for more methods on `RustwideBuilder`, which then means converting the errors each time when we call out of the class. 

I'll try to play around with some boundaries and then see if something looks better.

Reply:
If it's hard no worries, this is fine as-is.

Reply:
( there is a chance it's also just too much for my rust-skills, I saw parts of the approach in the [branch by Nemo](https://github.com/rust-lang/docs.rs/compare/master...Nemo157:anyhow))